core/bin/oauth.rs

```diff
@@ -203,6 +204,68 @@ async fn connections_access_token(
     }
 }
 
+#[derive(Deserialize)]
+struct CredentialPayload {
+    provider: CredentialProvider,
+    credentials: serde_json::Map<String, serde_json::Value>,
+}
+
+async fn oauth_service_create_credential(
+    State(state): State<Arc<OAuthState>>,
+    Json(payload): Json<CredentialPayload>,
+) -> (StatusCode, Json<APIResponse>) {
+    match Credential::create(state.store.clone(), payload.provider, payload.credentials).await {
+        Err(e) => error_response(
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "internal_server_error",
+            "Failed to create credential",
+            Some(e),
+        ),
+        Ok(c) => (
+            StatusCode::OK,
+            Json(APIResponse {
+                error: None,
+                response: Some(json!({
+                    "credential": {
+                        "credential_id": c.credential_id(),
+                        "created": c.created(),
+                        "provider": c.provider(),
+                        "credentials": c.credentials(),
+                    },
+                })),
+            }),
+        ),
+    }
+}
+
+async fn oauth_service_retrieve_credential(
```

Comment:
Should be `credentials_finalize`

Reply:
Why finalize ? There's nothing to finalize, so I don't understand the naming ? Should just be create / retrieve IMO.

Reply:
Sorry not finalize `credentials_retrieve`